pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -90,19 +90,15 @@ func (builder *OnBuild) Build(config *api.Config) (*api.Result, error) {
 	// Change the installation directory for this config to store scripts inside
 	// the application root directory.
 	if err := builder.source.Prepare(config); err != nil {
-		return buildResult, err
+		return nil, err
 	}
 
 	// If necessary, copy the STI scripts into application root directory
 	builder.copySTIScripts(config)
 
 	glog.V(2).Info("Creating application Dockerfile")
 	if err := builder.CreateDockerfile(config); err != nil {
-		buildResult.BuildInfo.FailureReason = utilstatus.NewFailureReason(
-			utilstatus.ReasonDockerfileCreateFailed,
-			utilstatus.ReasonMessageDockerfileCreateFailed,
-		)
-		return buildResult, err
```

Comment:
it seems like i'm introducing yet another thing that doesn't work. like the result.success field for example and i'm bloating the code for no particular reason.
if you think that we're going to use onbuild at some point in the near future or really want it to be there, then i revert the change.